core/src/com/unciv/ui/trade/DiplomacyScreen.kt

```diff
@@ -174,7 +174,16 @@ class DiplomacyScreen(val viewingCiv:CivilizationInfo):CameraStageBaseScreen() {
         }
         diplomacyTable.add(giveGiftButton).row()
         if (isNotPlayersTurn()) giveGiftButton.disable()
-        
+
+
+        val improveTileButton = "Gift Improvement".toTextButton()
```

Comment:
This should only appear in the first place if there are improvable tiles - meaning the function 'check if improvable tiles' should include the bonus resource filter

Reply:
fixed